homeassistant/components/gardena/vacuum.py

```diff
@@ -0,0 +1,174 @@
+
+    async def async_update(self):
+        """Update the states of Gardena devices."""
+        _LOGGER.debug("Running Gardena update")
+        # Managing state
+        state = self._device.state
+        _LOGGER.debug("Mower has state %s", state)
+        if state in ["WARNING", "ERROR", "UNAVAILABLE"]:
+            _LOGGER.debug("Mower has an error")
+            self._state = STATE_ERROR
+            self._error_message = self._device.last_error_code
+        else:
+            _LOGGER.debug("Getting mower state")
+            activity = self._device.activity
+            _LOGGER.debug("Mower has activity %s", activity)
+            if activity == "PAUSED":
+                self._state = STATE_PAUSED
+            elif activity in [
+                "OK_CUTTING",
+                "OK_CUTTING_TIMER_OVERRIDDEN",
+                "OK_LEAVING",
+            ]:
+                self._state = STATE_CLEANING
+            elif activity == "OK_SEARCHING":
+                self._state = STATE_RETURNING
+            elif activity in [
+                "OK_CHARGING",
+                "PARKED_TIMER",
+                "PARKED_PARK_SELECTED",
+                "PARKED_AUTOTIMER",
+            ]:
+                self._state = STATE_DOCKED
+            elif activity == "NONE":
+                _LOGGER.debug("Mower has no activity")
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._device.name
+
+    @property
+    def supported_features(self):
+        """Flag lawn mower robot features that are supported."""
+        return SUPPORT_GARDENA
+
+    @property
+    def battery_level(self):
+        """Return the battery level of the lawn mower."""
+        return self._device.battery_level
+
+    @property
+    def state(self):
+        """Return the status of the lawn mower."""
+        return self._state
+
+    @property
+    def available(self):
+        """Return True if the device is available."""
+        return self._device.state != "UNAVAILABLE"
+
+    def error(self):
+        """Return the error message."""
+        if self._state == STATE_ERROR:
+            return self._error_message
+        return ""
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the lawn mower."""
+        return {
+            "name": self._name,
+            "state": self._state,
+            "activity": self._device.activity,
+            ATTR_BATTERY_LEVEL: self._device.battery_level,
+            "battery_state": self._device.battery_state,
+            "rf_link_level": self._device.rf_link_level,
+            "rf_link_state": self._device.rf_link_state,
+            "serial": self._device.serial,
+            "operating_hours": self._device.operating_hours,
+        }
+
+    def start(self):
+        """Start the mower."""
+        self.turn_on()
+
+    def turn_on(self):
+        """Start cleaning or resume mowing."""
+        duration = 3600
```

Comment:
Default should be passed to the configuration validation as default.

Reply:
Right